internal/services/codesigning/trusted_signing_account_resource.go

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package codesigning
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/codesigning/2024-09-30-preview/codesigningaccounts"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type TruestedSigningAccountModel struct {
+	Name              string            `tfschema:"name"`
+	Location          string            `tfschema:"location"`
+	ResourceGroupName string            `tfschema:"resource_group_name"`
+	AccountUri        string            `tfschema:"account_uri"`
+	SkuName           string            `tfschema:"sku_name"`
+	Tags              map[string]string `tfschema:"tags"`
+}
+
+type TrustedSigningAccountResource struct{}
+
+var _ sdk.Resource = (*TrustedSigningAccountResource)(nil)
+
+func (m TrustedSigningAccountResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.All(
+				validation.StringLenBetween(3, 24),
+				validation.StringMatch(
+					regexp.MustCompile("^[A-Za-z][A-Za-z0-9]*(?:-[A-Za-z0-9]+)*$"),
+					"An account's name must be between 3-24 alphanumeric characters. The name must begin with a letter, end with a letter or digit, and not contain consecutive hyphens.",
+				),
+			),
+		},
+
+		"location": commonschema.Location(),
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"sku_name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ValidateFunc: validation.StringInSlice(
+				codesigningaccounts.PossibleValuesForSkuName(),
+				false),
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (m TrustedSigningAccountResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"account_uri": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (m TrustedSigningAccountResource) ModelObject() interface{} {
+	return &TruestedSigningAccountModel{}
+}
+
+func (m TrustedSigningAccountResource) ResourceType() string {
+	return "azurerm_trusted_signing_account"
+}
+
+func (m TrustedSigningAccountResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, meta sdk.ResourceMetaData) error {
+			client := meta.Client.CodeSigning.Client.CodeSigningAccounts
+
+			var model TruestedSigningAccountModel
+			if err := meta.Decode(&model); err != nil {
+				return err
```

Comment:
These errors should be wrapped
```suggestion
				return fmt.Errorf("decoding: %+v", err)
```